### PR TITLE
Pop out of Threads Activity Centre

### PR DESCRIPTION
**Pull request by florianduros**
## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

Closes https://github.com/element-hq/wat-internal/issues/144
[Figma design](https://www.figma.com/file/WGM39wEVQSJxnSFf6eeVry/Fixing-Threads-notifs?type=design&node-id=732-57806&mode=design&t=3Q92Nv3dV6HoGrPb-0)

This PR introduces a new way to interact with thread notifications. Following the previous PR https://github.com/matrix-org/matrix-react-sdk/pull/12137 which has added the labs flag and split the unread counts between thread and the main timeline. The threads unreads are now listed in the new **Threads Activity Centre** (TAC) which is located on the bottom left of Element Web, in the Spaces panel.

The TAC can be opened by clicking on the thread icon in the Spaces Panel:
- The dot color on the thread icon depicts the *level* of the unread in a thread. (Red for mention, green for a message...)
- It's merging all the *level* of all thread unreads. So if one thread has a mention, the icon will be red
- According to the user notification settings

The TAC is a pop out:
- All the rooms with thread unreads are listed
- Sorted by unread *levels* (`red > green > black/white`)
- A click on a room in the TAC, opens the room and the thread panel

## Examples

**Opening the TAC**
image

**Clicking on the room `test` in the TAC**
image

**Opening the thread with a mention in the room `test` and opening again the TAC. Both the TAC icon button and the room indicator have been updated to reflect the new state of the thread unreads**
image

## Included PRs

- https://github.com/matrix-org/matrix-react-sdk/pull/12156
- https://github.com/matrix-org/matrix-react-sdk/pull/12161
- https://github.com/matrix-org/matrix-react-sdk/pull/12179
- https://github.com/matrix-org/matrix-react-sdk/pull/12186
- https://github.com/matrix-org/matrix-react-sdk/pull/12199
- https://github.com/matrix-org/matrix-react-sdk/pull/12200
- https://github.com/matrix-org/matrix-react-sdk/pull/12207
- https://github.com/matrix-org/matrix-react-sdk/pull/12212
- https://github.com/matrix-org/matrix-react-sdk/pull/12226
- https://github.com/matrix-org/matrix-react-sdk/pull/12227

---
Here's what your changelog entry will look like:

## ✨ Features
 * Pop out of Threads Activity Centre ([\#12136](https://github.com/matrix-org/matrix-react-sdk/pull/12136)). Fixes element-hq/wat-internal#144. Contributed by @florianduros.

### CODE REVIEW

File: playwright/e2e/spaces/threads-activity-centre/index.ts [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+ * A utility that is able to find messages based on their content, by looking
+ * inside the `timeline` objects in the object model.
+ *
+ * Crucially, we hold on to references to events that have been edited or
+ * redacted, so we can still look them up by their old content.
+ *
+ * Provides utilities that build on the ability to find messages, e.g. replyTo,
+ * which finds a message and then constructs a reply to it.
+ */
+export class MessageBuilder {
```
Comment (florianduros) [inline playwright/e2e/spaces/threads-activity-centre/index.ts:71]: This is basically a lighter version of the read-receipt MessageBuilder.
Copy paste, added mention in thread and removed all the unused parts.

**Review by dbkr — APPROVED** [review]

File: src/Unread.ts [hunk 2]
```diff
@@ -105,6 +106,29 @@ function doesTimelineHaveUnreadMessages(room: Room, timeline: Array<MatrixEvent>
...
+export function doesRoomHaveUnreadThreads(room: Room): boolean {
+    if (getRoomNotifsState(room.client, room.roomId) === RoomNotifState.Mute) {
+        // No unread for muted rooms, nor their threads
+        // NB. This logic duplicated in RoomNotifs.determineUnreadState
+        return false;
+    }
+
+    for (const thread of room.getThreads()) {
+        if (doesTimelineHaveUnreadMessages(room, thread.timeline)) {
+            // We found an unread, so the room is unread
```
Comment (andybalaam) [inline src/Unread.ts:123]:
```suggestion
            // We found an unread, so the room has an unread thread
```

File: src/Unread.ts [hunk 3]
```diff
@@ -105,6 +106,29 @@ function doesTimelineHaveUnreadMessages(room: Room, timeline: Array<MatrixEvent>
...
+            return true;
+        }
+    }
+
+    // If we got here then no timelines were found with unread messages.
```
Comment (andybalaam) [inline src/Unread.ts:128]:
```suggestion
    // If we got here then no threads were found with unread messages.
```

**Review by andybalaam — APPROVED** [review]
Looks good. I especially like the Playwright tests.

2 tiny comment fixes.

**Pull request merged**